hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/RowSource.java

```diff
@@ -41,6 +41,9 @@ public RowSource(TypedProperties props, JavaSparkContext sparkContext, SparkSess
   @Override
   protected final InputBatch<Dataset<Row>> fetchNewData(Option<String> lastCkptStr, long sourceLimit) {
     Pair<Option<Dataset<Row>>, String> res = fetchNextBatch(lastCkptStr, sourceLimit);
+    if (overriddenSchemaProvider != null) {
+      return new InputBatch<>(res.getKey(), res.getValue(), overriddenSchemaProvider);
+    }
```

Comment:
`org.apache.hudi.utilities.sources.Source#fetchNext` actually checks and uses `overriddenSchemaProvider`. 
And `fetchNewData()` is only used in `fetchNext()` . i think some misconfig caused the issue.

Reply:
That's actually a valid point. I wrote a unit test with evolving schema. But, it passes even without this change. I think we can hold off landing this PR. Let me investigate more.